### DISCUSSION TITLE
Long passphrases are truncated on Unix-like by default

### DISCUSSION THREAD

**Discussion by hakavlad** [samuel-lucas6/Kryptor#79] [General]
Linux and macOS terminal emulators TRUNCATE input to 4095. Windows - not. This problem is a bit similar to the one we discussed recently.
https://stackoverflow.com/questions/18015137/linux-terminal-input-reading-user-input-from-terminal-truncating-lines-at-4095

In Linux, the command 
```
$ stty -icanon   # Disable canonical mode.
```
disables the input truncation.

Possible solution:
Truncate passphrases to 2048 bytes to ensure crossplatform compatibility. 
(and warn users that very long phrases will be truncated for compatibility)

**Comment by samuel-lucas6** [comment]
That's an interesting point. This will need some testing. I should've restricted them to 128 characters like in Cahir because there's no reason to use excessively long passphrases. Fortunately, nobody should realistically be affected by this.

**Reply by samuel-lucas6** [reply]
I quickly tested this on a Debian distro and the macOS Terminal, and there was no problem for both `-p:"password"` and `-p`.

**Reply by hakavlad** [reply]
Sorry, I didn't test this with Kryptor before reporting the issue. Kryptor is getting untruncated input for some reason. The issue has been identified and is specific to Python and `getpass.getpass()`. 
`age` also needs to be checked.

**Reply by hakavlad** [reply]
It looks like `age` has this problem. More testing is needed.
